File: OOPs/Inheritance/overriding_with_inheritance.py
class Parent:
    def __init__(self):
        self.x = 10
    
    def show(self):
        print("Parent's show()", self.x)

class Child(Parent):
    def show(self, x):
        print("Child's show()", self.x+x)

obj = Child()
obj.show(5)

File: OOPs/Inheritance/overriding_with_inheritance.cpp
#include<bits\stdc++.h>
using namespace std;

class Parent {
    public:
        int x;
        Parent() {
            x = 5;
        }
        virtual void showValue(int x) {
            cout << "Parent show() " << x << endl;
        }
};

class Child : public Parent {
    public:
        void showValue(int x) override {
            cout << "Child Show() " << this->x + x << endl; 
        }
};

int main() {
    Child obj;
    obj.showValue(10); // Output: Child Show() 15
}